unstructured/chainable.go

```diff
@@ -0,0 +1,30 @@
+package unstructured
+
+// Chainable is an interface to a more-easily chain Unstructured types.
+type Chainable interface {
+	// Data returns the data pointed by the current element, or the
+	// current saved error. The error might have been carried from a
+	// previous failure to reach or change an element.
+	//
+	// Note that the first error will preempt over all other
+	// operations. In other words, if you chain multiple set
+	// operations, only the ones before the first failure will happen,
+	// others will be ignored.
+	Data() (interface{}, error)
+
+	// Field returns a Chainable to the object below the "key" field.
```

Comment:
What does "below" mean?

What if teh key's value is an int?

Reply:
I'll explain that specific use case, and add an extra-example to show how this works.